test/ng/compileSpec.js

```diff
@@ -3373,6 +3373,90 @@ describe('$compile', function() {
     });
   });
 
+  describe('collector', function() {
+
+    var collected;
+    beforeEach(module(function($compileProvider) {
+      collected = false;
+      $compileProvider.directive('testCollect', function() {
+        return {
+          restrict: 'EACM',
+          link: function() {
+            collected = true;
+          }
+        };
+      });
+    }));
+
+    it('should collect comment directives by default', inject(function() {
+      var html = '<!-- directive: test-collect -->';
+      element = $compile('<div>' + html + '</div>')($rootScope);
+      expect(collected).toBe(true);
+    }));
+
+    it('should collect css class directives by default', inject(function() {
+      element = $compile('<div class="test-collect"></div>')($rootScope);
+      expect(collected).toBe(true);
+    }));
+
+    angular.forEach([
+      { commentEnabled: true, cssEnabled: true },
+      { commentEnabled: true, cssEnabled:false },
+      { commentEnabled:false, cssEnabled: true },
+      { commentEnabled:false, cssEnabled:false }
```

Comment:
Ahhh, ok!

On Wed, Jul 27, 2016 at 12:00 PM Georgios Kalpakas <notifications@github.com>
wrote:

> In test/ng/compileSpec.js
> <https://github.com/angular/angular.js/pull/14850#discussion_r72412163>:
>
> > +    it('should collect comment directives by default', inject(function() {
> > +      var html = '<!-- directive: test-collect -->';
> > +      element = $compile('<div>' + html + '</div>')($rootScope);
> > +      expect(collected).toBe(true);
> > +    }));
> > +
> > +    it('should collect css class directives by default', inject(function() {
> > +      element = $compile('<div class="test-collect"></div>')($rootScope);
> > +      expect(collected).toBe(true);
> > +    }));
> > +
> > +    angular.forEach([
> > +      { commentEnabled: true, cssEnabled: true },
> > +      { commentEnabled: true, cssEnabled:false },
> > +      { commentEnabled:false, cssEnabled: true },
> > +      { commentEnabled:false, cssEnabled:false }
>
> @petebacondarwin <https://github.com/petebacondarwin>, just trying to
> proactively protect @mgol <https://github.com/mgol>'s PR from getting
> stale 😛
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/angular/angular.js/pull/14850/files/f248bd71de405faeca5056e2898d2f924708992d#r72412163>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACoWydUjkPuAVP74kJE6L-cR5geprGoeks5qZywegaJpZM4JB9-c>
> .
>
-- 
Dr. David Ródenas
*Vice-degà *
Col·legi Oficial d’Enginyeria Informàtica de Catalunya